src/SFA.DAS.PR.Application.UnitTests/Requests/Commands/CreateAddAccountRequest/CreateAddAccountRequestCommandValidatorTests.cs

```diff
@@ -0,0 +1,81 @@
+            _accountLegalEntityReadRepositoryValidMock.Object
+        );
+        var result = await sut.TestValidateAsync(new CreateAddAccountRequestCommand { Ukprn = 10000003, AccountLegalEntityId = 1, RequestedBy = Guid.NewGuid().ToString(), EmployerContactEmail = "test@email.com", Operations = [Operation.CreateCohort] });
+        result.ShouldNotHaveValidationErrorFor(query => query.Ukprn);
+        result.ShouldNotHaveValidationErrorFor(query => query.AccountLegalEntityId);
+        result.ShouldNotHaveValidationErrorFor(query => query.EmployerContactEmail);
+        result.ShouldNotHaveValidationErrorFor(query => query.Ukprn);
+    }
+
+    [Test]
+    public async Task CreateAddAccountRequestCommand_Invalid_Ukprn()
+    {
+        var sut = new CreateAddAccountRequestCommandValidator(
+            _requestReadRepositoryValidMock.Object,
+            _providersReadRepositoryValidMock.Object,
+            _accountLegalEntityReadRepositoryValidMock.Object
+        );
+        var result = await sut.TestValidateAsync(new CreateAddAccountRequestCommand { Ukprn = 1000000, AccountLegalEntityId = 1, RequestedBy = Guid.NewGuid().ToString(), EmployerContactEmail = "test@email.com", Operations = [Operation.CreateCohort] });
+        result.ShouldHaveValidationErrorFor(q => q.Ukprn)
+            .WithErrorMessage(UkprnValidator.UkprnFormatValidationMessage);
+    }
+
+    [Test]
+    public async Task CreateAddAccountRequestCommand_Invalid_AccountLegalEntityId()
+    {
+        var sut = new CreateAddAccountRequestCommandValidator(
+            _requestReadRepositoryValidMock.Object,
+            _providersReadRepositoryValidMock.Object,
+            _accountLegalEntityReadRepositoryInvalidMock.Object
+        );
+        var result = await sut.TestValidateAsync(new CreateAddAccountRequestCommand { Ukprn = 10000002, AccountLegalEntityId = 0, RequestedBy = Guid.NewGuid().ToString(), EmployerContactEmail = "test@email.com", Operations = [Operation.CreateCohort] });
+        result.ShouldHaveValidationErrorFor(q => q.AccountLegalEntityId)
+            .WithErrorMessage(AccountLegalEntityValidator.AccountLegalEntityIdValidationMessage);
+    }
+
+    [Test]
+    public async Task CreateAddAccountRequestCommand_Invalid_EmployerContactEmail()
+    {
+        var sut = new CreateAddAccountRequestCommandValidator(
+            _requestReadRepositoryValidMock.Object,
+            _providersReadRepositoryValidMock.Object,
+            _accountLegalEntityReadRepositoryValidMock.Object
+        );
+        var result = await sut.TestValidateAsync(new CreateAddAccountRequestCommand { Ukprn = 10000002, AccountLegalEntityId = 1, RequestedBy = Guid.NewGuid().ToString(), EmployerContactEmail = "testemail.com", Operations = [Operation.CreateCohort] });
+        result.ShouldHaveValidationErrorFor(q => q.EmployerContactEmail)
+            .WithErrorMessage(CreateAddAccountRequestCommandValidator.EmployerContactEmailValidationMessage);
+    }
+}
```

Comment:
You only have unit tests for some, but not all, of the validation rules. Am I correct in thinking that the validator base class has unit tests covering the validation rules not tested here?

Reply:
@PeterDavidWilliams Yea each validation specific class i also unit tested in its own right